starknet_py/net/full_node_client.py

```diff
@@ -101,6 +104,79 @@ async def get_block_traces(
     ) -> BlockTransactionTraces:
         raise NotImplementedError()
 
+    async def get_events(
+        self,
+        *,
+        address: Hash,
+        keys: List[str],
+        from_block_number: Optional[Union[int, Tag]] = None,
+        from_block_hash: Optional[Union[Hash, Tag]] = None,
+        to_block_number: Optional[Union[int, Tag]] = None,
+        to_block_hash: Optional[Union[Hash, Tag]] = None,
+        follow_continuation_token: bool = False,
+        chunk_size: int = 1,
+    ) -> EventsResponse:
+        # pylint: disable=too-many-arguments
+        """
+        :param address: The address of the contract that emitted the event.
+            Must be in a form of RPC accepted felt.
+        :param keys: List of names of events that are searched for.
+            Must be in a form of RPC accepted felt after being hashed by `keccak` hash.
+        :param from_block_number: Number of the block from which events searched for **starts**
+            or literals `"pending"` or `"latest"`. Mutually exclusive with ``from_block_hash`` parameter.
+        :param from_block_hash: Hash of the block from which events searched for **starts**
+            or literals `"pending"` or `"latest"`. Mutually exclusive with ``from_block_number`` parameter.
+        :param to_block_number: Number of the block to which events searched for **end**
+            or literals `"pending"` or `"latest"`. Mutually exclusive with ``to_block_hash`` parameter.
+        :param to_block_hash: Hash of the block to which events searched for **end**
+            or literals `"pending"` or `"latest"`. Mutually exclusive with ``to_block_number`` parameter.
+        :param follow_continuation_token: Flag deciding whether all events should be collected during one function call,
+                defaults to False.
+        :param chunk_size: Size of chunk of events returned by one ``get_events`` call, defaults to 1 (minimum).
+
+        :return: ``EventsResponse`` dataclass containing events and optional continuation token.
+        """
+
+        if chunk_size <= 0:
+            raise ValueError("Argument chunk_size must be grater than 0.")
+        params = {
+            "chunk_size": chunk_size,
+            "from_block": get_small_block_identifier(
+                from_block_hash, from_block_number
+            ),
+            "to_block": get_small_block_identifier(to_block_hash, to_block_number),
+            "address": address,
```

Comment:
`address` parameter is typed as `Hash`, which is `Union[int, str]` under the hood. Can `address` field be an integer? Or should it be felt?

Reply:
You're right, according to specification it should be a felt, but in RPC form. Is leaving `str` as a type fine or should I make an alias for `str`, something like `RPCFelt`?

Reply:
I think leaving `address` typed as `Hash` is fine. We want the method to be flexible for users. What we should do is parse the address and have it as a proper RPC felt at this point in code.